Linkit / EV3 howto

cavedunissin opened the pull request:
update LinkIt / Lego EV3 tutorials, code and HowTos entries. (Hope there will no errors) - David.

cavedunissin committed:
Add files via upload

tutorial update

cavedunissin committed:
Add files via upload

add analogread tutorial of LinkIt 7697

cavedunissin committed:
Add files via upload

source code of LinkIt 7697 analogread tutorial

cavedunissin committed:
Add files via upload

add EV3 tutorial source code

cavedunissin committed:
Add files via upload

tutorial of Lego EV3 robot - button control

cavedunissin committed:
Update MicroBitButton.md

cavedunissin committed:
Update MicroBitMagnetometer.md

add tutorial/code link

cavedunissin committed:
Delete EV3_ButtonControl.pdf

cavedunissin committed:
Add files via upload

add Lego EV3 robot tutorial - Button control

cavedunissin committed:
Add files via upload

add LinkIt 7697, 7688, Lego EV3 icon

cavedunissin committed:
Delete MIT_App_Inventor_analogread_7697.pdf

cavedunissin committed:
Add files via upload

add LinkIt 7697 tutorial - LED, analogread

cavedunissin committed:
Delete MIT_App_Inventor_ButtonControl_EV3.pdf

cavedunissin committed:
Update HowTos.js

ewpatton commented:
Thanks for the pull request, David. GitHub is reporting that there are two convicts with master. Would you please either merge master into this branch or rebase this on top of master to resolve the conflicts?

cavedunissin committed:
Add linkit tutorials

cavedunissin committed:
Merge branch 'master' into linkit-howto

cavedunissin committed:
tutorials upload

upload tutorials and source code of Lego EV3 (button control and sensor panel), LinkIt 7697 (BLE LED and mositure sensor)

cavedunissin committed:
upload tutorials

upload new version of Micro:bit button and magentometer tutorial

cavedunissin committed:
Merge pull request #161 from cavedunissin/linkit-howto

upload tutorials

cavedunissin committed:
update

cavedunissin committed:
Merge pull request #162 from cavedunissin/linkit-howto

update micro:bit tutorial

ewpatton commented on docs/makepages.py:
```diff
@@ -100,7 +100,14 @@
     </Route>
-    <Route path="faq">
+    <Route path="linkit">
+	  <Route path="linkitintro" component={linkitintroPage} />
```
After compiling, the page doesn't load here due to capitalization mismatch between the `...Page` components and the file names in the pages directory.

ewpatton commented on docs/src/app/HowTos.js:
```diff
@@ -142,10 +142,42 @@ const howtosList = [
+  },{
+    title: 'How to: Read analog data',
+    author: 'MIT App Inventor Project',
+    img: 'assets/howtos/LinkIt_7697.png',
```
Verify these image files. They don't show up if I fix the errors above.

ewpatton commented on docs/src/app/pages/LinkIt7688.md:
```diff
@@ -0,0 +1,27 @@
+# Platform: Arduino 101
```
This file is just a copy of the Arduino 101 page. Please update it before we can finish reviewing.

ewpatton commented on docs/src/app/pages/LinkIt7697.md:
```diff
@@ -0,0 +1,27 @@
+# Platform: Arduino 101
```
Ditto this being a copy.

ewpatton commented on docs/src/app/pages/LinkItintro.md:
```diff
@@ -0,0 +1,33 @@
+
+The following extensions are available for use with the companion sketch as part of the [Arduino 101 Extension Bundle](/assets/resources/edu.mit.appinventor.iot.arduino101.aix):
+
+* [Arduino 101 Accelerometer](#/arduino101/arduinoaccelerometer) &mdash; Access the onboard digital 3-axis accelerometer on the Curie's inertial measurement unit (IMU) chip.
```
Do all of the Arduino 101 extensions work with the LinkIt? Otherwise you should update this list.

ewpatton reviewed (changes requested):
Hi David. Please see the feedback I left. At the moment if I cleanly check out this branch and try to run it there are issues w.r.t. file name case and image paths.

cavedunissin committed:
update

update tutorials/codes/md file of Lego EV3/LinkIt 7697/Micro:bit

cavedunissin committed:
tutorial update

cavedunissin committed:
Merge pull request #163 from cavedunissin/linkit-howto

update tutorials/md/code of Lego, LinkIt and microbit

ewpatton reviewed (changes requested):
@cavedunissin You should also update the top navigation and the devices page to include entries for the LEGO and LinkIt. I would also suggest putting the LEGO entry before LinkIt to keep the list sorted alphabetically.

ewpatton commented:
Take a look at Master.js (to handle page titles), DocumentationMenu.js (to add to top menu), and DeviceList.js (to add to device table).

cavedunissin committed:
update

cavedunissin committed:
update

cavedunissin committed:
Merge pull request #164 from cavedunissin/linkit-howto

Linkit howto

ewpatton commented on docs/src/app/HowTos.js:
```diff
@@ -141,12 +141,71 @@ const howtosList = [
+    equipment: ['LinkIt 7697', 'RGB LED'],
+    time: 90,
+  },
+  /*{
```
I'd prefer you not include commented out entries. Add them when they are ready.

ewpatton commented on docs/src/app/HowTos.js:
```diff
@@ -141,12 +141,71 @@ const howtosList = [
+    equipment: ['Lego EV3 robot kit'],
+    time: 60,
+  },
+  /*{
```
See comment above about commented-out code.

ewpatton commented on docs/src/app/Master.js:
```diff
@@ -211,6 +212,7 @@ class Master extends Component {
               <ForStudentsMenu />
               <ForMakersMenu />
               <DocumentationMenu />
+			  <haha />
```
What is this?

ewpatton commented on docs/src/app/pages/LinkIt7697.md:
```diff
@@ -0,0 +1,35 @@
+
+Please follow our instructions:
+
+* [LED control (BluetoothLE)](/assets/howtos/MIT_App_Inventor_7697_LED.pdf) &mdash; Control the onboard LED by button and voice through Bluetooth.
```
The tutorials for Arduino and micro:bit open in a new tab. You should look at those files for an example so that the behavior is the same.

ewpatton commented on docs/src/app/pages/legoev3intro.md:
```diff
@@ -0,0 +1,26 @@
+</div>
+
+MIT App Inventor provides complete support to interact with Lego EV3, including:
+* [Ev3Motors](http://ai2.appinventor.mit.edu/reference/components/legomindstorms.html#Ev3Motors) &mdash; control motor speed and direction.
```
Ditto above about opening external links in a new tab.

ewpatton commented on docs/src/app/pages/legoev3intro.md:
```diff
@@ -0,0 +1,26 @@
+
+Please check tutorials:
+
+* [Button Control](assets/howtos/MIT_App_Inventor_EV3_ButtonControl.pdf) &mdash; Control Lego EV3 robot with buttons. You can also adjust the motor speed by sliders..
```
Open link in a new tab.

ewpatton reviewed (changes requested):
@cavedunissin Almost there. Please address the comments I made and then we can put this up on the test server.

ewpatton commented:
@cavedunissin You should also merge the master branch from mit-cml into your branch and resolve the conflict on the Magnetometer PDF file.

cavedunissin committed:
update

modify open link in new tab and type error

cavedunissin committed:
Merge pull request #165 from cavedunissin/linkit-howto

modify open link in new tab and type error

ewpatton commented on docs/src/app/pages/LinkIt7697.md:
```diff
@@ -0,0 +1,34 @@
+* Open hardware board reference design including schematic, <a href="https://labs.mediatek.com/en/download/1ega2lbl" target="_blank">layouts and pin details.</a>
+
+Please follow our instructions:
+<ul><li><a href='http://iot.appinventor.mit.edu/assets/howtos/MIT_App_Inventor_7697_LED.pdf' target='_blank'>LED Control (BluetoothLE)</a> &mdash; Control the onboard LED by button and voice through Bluetooth.</li></ul>
```
Rather than providing an absolute URL, use a relative URL starting with `/`, for example `/assets/howtos/MIT_App_Inventor_7697_LED.pdf`. Right now all of these links are broken when testing locally because the URLs point to the production site.

ewpatton replied:
Rather than having a `<ul>` per line, you should start with a `<ul>` and only put the `</ul>` at the end of the list. As given, you are creating a new list per line rather than the more desirable multiple `<li>` in a single list.

ewpatton commented on docs/src/app/pages/MicroBitAccelerometer.md:
```diff
@@ -4,6 +4,10 @@ The <code>Microbit_Accelerometer</code> component lets users configure the BBC m
 The accelerometer's report rate, or period, determines how frequently data will be sent to App Inventor.
 
+<strong>More links:</strong><ul><li>Download a <a href='http://iot.appinventor.mit.edu/assets/samples/MicrobitAccelerometer.aia' target='_blank'>sample projects</a> for the micro:bit onboard buttons.</li>
+                                <li>View the <a href='http://iot.appinventor.mit.edu/assets/howtos/MIT_App_Inventor_Microbit_Accelerometer.pdf' target='_blank'>how to instructions</a> for the micro:bit onboard buttons</li></ul>
```
Remove the extraneous whitespace at the start of this line.

ewpatton commented on docs/src/app/pages/MicroBitButton.md:
```diff
@@ -3,6 +3,9 @@
 <ul><li>0 - up</li><li>1 - down</li><li>2 - long pressed</li></ul>
 
+<strong>More links:</strong><ul><li>Download a <a href='http://iot.appinventor.mit.edu/assets/samples/MicrobitButton.aia' target='_blank'>sample projects</a> for the micro:bit onboard buttons.</li>
+                                <li>View the <a href='http://iot.appinventor.mit.edu/assets/howtos/MIT_App_Inventor_Microbit_Button.pdf' target='_blank'>how to instructions</a> for the micro:bit onboard buttons</li></ul>
```
Ditto whitespace.

ewpatton commented on docs/src/app/pages/MicroBitDeviceInformation.md:
```diff
@@ -1,6 +1,10 @@
+The <code>Microbit_Device_Information</code> extension provides App Inventor with information about Microbit, including Firmware version, Manufacturer name and serial nuber, etc.
+
+<strong>More links:</strong><ul><li>Download a <a href='http://iot.appinventor.mit.edu/assets/samples/MicrobitDeviceInformation.aia' target='_blank'>sample projects</a> for the micro:bit onboard buttons.</li>
+                                <li>View the <a href='http://iot.appinventor.mit.edu/assets/howtos/MIT_App_Inventor_Microbit_DeviceInformation.pdf' target='_blank'>how to instructions</a> for the micro:bit onboard buttons</li></ul>
```
Ditto whitespace.

ewpatton commented on docs/src/app/pages/MicroBitIoPin.md:
```diff
@@ -1,7 +1,7 @@
 
 The <code>Microbit_Io_Pin</code> component lets users configure the BBC micro:bit's analog pins for input and output, and to read, write, and request notifications for the I/O pin states.
-
+ 
```
Random space added to this line should be removed.

ewpatton commented on docs/src/app/pages/MicroBitMagnetometer.md:
```diff
@@ -2,6 +2,9 @@
 The <code>Microbit_Magnetometer</code> provides the ability to configure the BBC micro:bit's on-board magnetometer and receive one or more magnetometer samples via the appropriate methods.
 
+<strong>More links:</strong><ul><li>Download a <a href='http://iot.appinventor.mit.edu/assets/samples/MicrobitMagnetometer.aia' target='_blank'>sample projects</a> for the micro:bit onboard magnetometer.</li>
+                                <li>View the <a href='http://iot.appinventor.mit.edu/assets/howtos/MIT_App_Inventor_Microbit_Magnetometer.pdf' target='_blank'>how to instructions</a> for the micro:bit onboard magnetometer</li></ul>
```
Ditto whitespace.

ewpatton commented on docs/src/app/pages/legoev3intro.md:
```diff
@@ -0,0 +1,25 @@
+</div>
+
+MIT App Inventor provides complete support to interact with Lego EV3, including:
+<ul><li><a href='http://ai2.appinventor.mit.edu/reference/components/legomindstorms.html#Ev3Motors' target='_blank'>Ev3Motors</a> &mdash; control motor speed and direction.</li></ul>
```
See my previous note about `<ul>`, `</ul>`, and `<li>` for this list.

ewpatton commented on docs/src/app/pages/legoev3intro.md:
```diff
@@ -0,0 +1,25 @@
+<ul><li><a href='http://ai2.appinventor.mit.edu/reference/components/legomindstorms.html#Ev3Commands' target='_blank'>Ev3Commands</a> &mdash; low level commands can talk to Ev3 brick directly.</li></ul>
+
+Please check Lego EV3 tutorials:
+<ul><li><a href='http://iot.appinventor.mit.edu/assets/howtos/MIT_App_Inventor_EV3_ButtonControl.pdf' target='_blank'>Button Control</a> &mdash; Control Lego EV3 robot with buttons. You can also adjust the motor speed by sliders.</li></ul>
```
See my previous note about relative vs absolute URLs and about `<ul>`, `</ul>`, and `<li>`.

ewpatton commented on docs/src/www/assets/samples/WIFI_7697_wheelrobot.ino.ino:
```diff
@@ -0,0 +1,247 @@
+/*
```
This file has two .ino extensions. It might be confusing to people, so you should rename the file with only a single .ino.

ewpatton reviewed (dismissed):
@cavedunissin Please check my comments about the changes you made in response to the last review. In particular be aware of the semantics of `<ul>` and also try to use relative URLs for the IOT site where possible otherwise it makes testing difficult.

cavedunissin committed:
update

update md files' <ul> issues with Evan's suggestions.

cavedunissin committed:
Merge pull request #166 from cavedunissin/linkit-howto

update

ewpatton commented on docs/src/app/pages/MicroBitIoPin.md:
```diff
@@ -1,6 +1,6 @@
 # Microbit\_Io\_Pin
 
-The <code>Microbit_Io_Pin</code> component lets users configure the BBC micro:bit's analog pins for input and output, and to read, write, and request notifications for the I/O pin states.
+The <code>Microbit_Io_Pin</code> component lets users configure the BBC micro:bit's analog pins for input and output, and to read, write, and request notifications for the I/O pin states.<br>
```
Why did you add a `<br>` here where there wasn't one previously?

cavedunissin replied:
Evan, I refer to MicroBitLed.md  to compare the difference

ewpatton reviewed (commented):
See comment

cavedunissin committed:
update

update MicroBitIoPin.md

cavedunissin committed:
Merge pull request #167 from cavedunissin/linkit-howto

update

ewpatton commented:
When I am on campus tomorrow I'll push this up to the test server so that we can have others try it out.

cavedunissin committed:
Committing for merge

cavedunissin committed:
update

cavedunissin committed:
update

cavedunissin committed:
Merge pull request #168 from cavedunissin/linkit-howto

update with linkit and mostly micro:bit

cavedunissin committed:
update 7697 LED tutorial

cavedunissin committed:
Merge pull request #169 from cavedunissin/linkit-howto

update 7697 LED tutorial

cavedunissin committed:
Update of MIT App Inventor Codi Bot

Add codibot page, tutorial and source code.

cavedunissin committed:
update

cavedunissin committed:
Merge pull request #172 from cavedunissin/linkit-howto

Update Codi Bot

cavedunissin committed:
update Codi Bot

cavedunissin committed:
Merge pull request #173 from cavedunissin/linkit-howto

update Codi Bot

ewpatton commented:
@cavedunissin I think it would be good to also include the Codi Bot in the devices table. It's only listed in the text at the moment.

ewpatton commented on docs/src/app/pages/codibotintro.md:
```diff
@@ -0,0 +1,35 @@
+
+</div>
+
+## Parts list (<a href="https://www.robotkingdom.com.tw/product/mit-app-inventor-codi-bot/" target="_blank">Purchase link)</a>:
```
Move the closing parenthesis outside of the `</a>`.

cavedunissin replied:
ok

ewpatton commented on docs/src/app/pages/codibotintro.md:
```diff
@@ -0,0 +1,35 @@
+## Parts list (<a href="https://www.robotkingdom.com.tw/product/mit-app-inventor-codi-bot/" target="_blank">Purchase link)</a>:
+* <a href="https://www.robotkingdom.com.tw/product/linkit-7697/" target="_blank">LinkIt 7697 MCU board</a>, 1
+* <a href="https://www.robotkingdom.com.tw/product/linkit-7697%E6%A9%9F%E5%99%A8%E4%BA%BA%E6%93%B4%E5%85%85%E6%9D%BF-robot-shield-for-linkit-7697/" target="_blank">Robot Shield for LinkIt 7697 V2.0</a>, 1 (with an onboard buzzer) 
+* RGB LED (common cathode), 1
```
Are we intending to link these other components? It seems odd to me that only the first two are linked.

cavedunissin replied:
Cuz the first two are the core component of Codi Bot, other are quite common and easy to get. You suggest I link all of them or remove the link of first two?

ewpatton commented on docs/src/app/pages/codibotintro.md:
```diff
@@ -0,0 +1,35 @@
+Compared with [LinkIt 7697 tutorials](#/linkit/linkit7697), tutorials for the MIT App Inventor Codi Bot use an extension of the LinkIt 7697 dev board instead of low-level Bluetooth commands.
+
+<ul>
+<li><a href='/assets/howtos/MIT_App_Inventor_CodiBot_UserManual.pdf' target='_blank'>User manual</a> &mdash; Follow this manual to assemble the MIT App Inventor Codi Bot.</li>
```
Has Hedge reviewed the PDFs for spelling and grammar?

ewpatton replied:
Also, is there a Google Doc somewhere, or something similar? I have some comments on the PDFs as well that I'd like to share.

cavedunissin replied:
no problem. mail to you

ewpatton commented on docs/src/app/pages/codibotintro.md:
```diff
@@ -0,0 +1,35 @@
+
+## Please follow our tutorials to get started:
+
+Compared with [LinkIt 7697 tutorials](#/linkit/linkit7697), tutorials for the MIT App Inventor Codi Bot use an extension of the LinkIt 7697 dev board instead of low-level Bluetooth commands.
```
I'm not sure that we need to justify/explain this here. I think the heading should be "Instructions and Samples" and then you can in the paragraph say "Once you have received your kit, please follow our tutorials to get started."

ewpatton commented on docs/src/www/assets/samples/BLE_7697_BLE_WalkingRobot.ino:
```diff
@@ -0,0 +1,201 @@
+/* To control OTTO's wlkaing and light with MIT App Inventor BLE and turn */
```
OTTO => Codi

ewpatton reviewed (changes requested):
See the comments I left. Also, please remove the Thumbs.db files and add it to your .gitignore.

cavedunissin committed:
update Codi Bot tutorials

update Codi Bot tutorials with Evan/Hedge comments

cavedunissin committed:
Merge pull request #176 from cavedunissin/linkit-howto

update Codi Bot tutorials

cavedunissin committed:
update Codi Bot tutorials

cavedunissin committed:
Revert "update Codi Bot tutorials"

This reverts commit c603b09163e71b486fe18319d9763999f42ec968.

cavedunissin committed:
update again

cavedunissin committed:
Merge pull request #177 from cavedunissin/linkit-howto

Linkit howto

cavedunissin committed:
update codibot.aia

cavedunissin committed:
Merge pull request #178 from cavedunissin/linkit-howto

update codibot.aia

cavedunissin committed:
update Codibot tutorials

cavedunissin committed:
Merge pull request #179 from cavedunissin/linkit-howto

update Codibot tutorials
